src/utilities/activities.ts

```
@@ -191,90 +191,89 @@ export function copyActivityDirectivesToClipboard(sourcePlan: Plan, activities:
+  const noun = `Activity Directive${activities.length === 1 ? '' : 's'}`;
+  setClipboardContent(
+    clipboard,
+    () => showSuccessToast(`Copied ${activities.length} ${noun}`),
+    () => showFailureToast(`Failed to copy ${activities.length} ${noun}`),
+  );
 }
 
-export function getPasteActivityDirectivesText(): string | undefined {
-  try {
-    const serializedClipboard = getSessionStorageClipboard();
-    if (serializedClipboard !== null) {
-      const clipboard = JSON.parse(serializedClipboard);
-      if (Array.isArray(clipboard.activities)) {
-        const n = clipboard.activities.length;
-        return `Paste ${n} Activity Directive${n === 1 ? '' : 's'}`;
-      }
-    }
-  } catch (e) {
-    console.error('e');
+export function getPasteActivityDirectivesText(count: number): string {
+  if (count <= 0) {
+    return `Paste Activity Directives`; //generic text, disabled context menu
+  } else {
+    return `Paste ${count} Activity Directive${count === 1 ? '' : 's'}`;
   }
 }
 
-export function canPasteActivityDirectivesFromClipboard(destinationPlan: Plan | null): boolean {
-  if (destinationPlan === null) {
-    return false;
-  }
-
+export async function getActivityDirectivesClipboardCount(): Promise<number> {
   try {
-    const serializedClipboard = getSessionStorageClipboard();
-    if (serializedClipboard === null) {
-      return false;
-    }
-
-    const clipboard = JSON.parse(serializedClipboard);
-    //current scope, allows copy/paste in the same plan
-    if (clipboard.sourcePlanId !== destinationPlan.id) {
-      return false;
+    const clipboardContent = await getClipboardContent();
+    if (clipboardContent !== undefined) {
+      const clipboard = JSON.parse(clipboardContent);
+      if (clipboard.type === 'aerie_activity_directives' && clipboard.activities !== undefined) {
+        return clipboard.activities.length;
+      }
     }
   } catch (e) {
-    console.error(e);
-    return false;
+    //eh silent
```

Any reason not to spit out a log here?

---

I'll spit a warning here. There's an error that happens sometimes the clipboard when page is not in focus, and i was trying to avoid dirtying up the log.

---

Actually i'll keep this silent. This throws an error if the clipboard has something that is not json, general clipboard things. It should fail silently.

---

Oh makes sense. Could you update the comment there with a note about that so that future us/others don't throw a log there?